### ISSUE TITLE
Missing methods for congruences

### ISSUE THREAD

**Issue by james-d-mitchell** [semigroups/Semigroups#370]
It is currently not possible to run `EquivalenceRelationLookup` nor `MeetSemigroupCongruences` for all types of congruences over a semigroup. I will post examples later when I have more time.

Here is an example: 

    gap> S := FullTransformationMonoid(3);;
    gap> C := CongruencesOfSemigroup(S);;
    gap> MeetSemigroupCongruences(C[1], C[1]);
    Error, no method found! For debugging hints type ?Recovery from NoMethodFound
    Error, no 1st choice method found for `MeetMagmaCongruences' on 2 arguments at /Users/jdm/gap/lib/methsel2.g:241 called from
    <function "HANDLE_METHOD_NOT_FOUND">( <arguments> )
     called from read-eval loop at *stdin*:126
    you can 'quit;' to quit to outer loop, or
    you can 'return;' to continue
    brk>

**Comment by tomcontileslie** [comment]
Related issue: the elements of a quotient semigroup `S/cong` currently do not satisfy  `IsAssociativeElement`.

**Comment by james-d-mitchell** [comment]
Thanks @tomcontileslie, it'd be extra helpful to have a minimum example of this to add to future tests!

**Comment by tomcontileslie** [comment]
Sorry I didn't provide one, here's an example:
```
gap> T := FullTransformationMonoid(2);;
gap> cong := SemigroupCongruence(T, []);;                           
gap> IsAssociativeElement(Elements(T/cong)[1]);
false 
```

**Comment by james-d-mitchell** [comment]
Thanks @tomcontileslie, most helpful!

**Comment by james-d-mitchell** [comment]
This was resolved in v4.0.0
